lib/wasmex.ex

```diff
@@ -92,6 +92,35 @@ defmodule Wasmex do
   - `:f32` a 32 bit float
   - `:f64` a 64 bit float
 
+  ### Linking
+
+  Wasm links may be given as an additional option.
+  Links is an array of module name-bytes pair.
+
+      iex> calculator_wasm = File.read!(TestHelper.wasm_link_test_file_path())
+      iex> utils_wasm = File.read!(TestHelper.wasm_test_file_path())
+      iex> links = %{utils: %{bytes: utils_wasm}}
+      iex> {:ok, pid} = Wasmex.start_link(%{bytes: calculator_wasm, links: links})
+      iex> Wasmex.call_function(pid, "sum_range", [1, 5])
+      {:ok, [15]}
+
+  It is also possible to link a module that was compiled for improved performance:
+
+      iex> calculator_wasm = File.read!(TestHelper.wasm_link_test_file_path())
+      iex> utils_wasm = File.read!(TestHelper.wasm_test_file_path())
+      iex> {:ok, store} = Wasmex.Store.new()
+      iex> {:ok, utils_module} = Wasmex.Module.compile(store, utils_wasm)
+      iex> links = %{utils: %{module: utils_module}}
+      iex> {:ok, pid} = Wasmex.start_link(%{bytes: calculator_wasm, links: links, store: store})
+      iex> Wasmex.call_function(pid, "sum_range", [1, 5])
+      {:ok, [15]}
+
+  **Notes:**
+  1. Make sure to use the same store for the linked modules and the main module.
+  2. This library does not currently support dependencies between linked modules.
+     For example: if Module A links to Module B, and Module B links to the Main module, this will not work.
+     Workaround: Link both Module A and B to the Main module directly.
+
```

Comment:
Nitpicking a bit on the docs (e.g. explaining why and when linking precompiled modules yields perf improvements).

```suggestion
  ### Linking multiple Wasm modules

  Wasm module `links` may be given as an additional option.
  Links is an array of Wasm module name-bytes pair.

      iex> calculator_wasm = File.read!(TestHelper.wasm_link_test_file_path())
      iex> utils_wasm = File.read!(TestHelper.wasm_test_file_path())
      iex> links = %{utils: %{bytes: utils_wasm}}
      iex> {:ok, pid} = Wasmex.start_link(%{bytes: calculator_wasm, links: links})
      iex> Wasmex.call_function(pid, "sum_range", [1, 5])
      {:ok, [15]}

  It is also possible to link a precompiled module.
  This improves performance if the same module is used many times by compiling it only once.

      iex> calculator_wasm = File.read!(TestHelper.wasm_link_test_file_path())
      iex> utils_wasm = File.read!(TestHelper.wasm_test_file_path())
      iex> {:ok, store} = Wasmex.Store.new()
      iex> {:ok, utils_module} = Wasmex.Module.compile(store, utils_wasm)
      iex> links = %{utils: %{module: utils_module}}
      iex> {:ok, pid} = Wasmex.start_link(%{bytes: calculator_wasm, links: links, store: store})
      iex> Wasmex.call_function(pid, "sum_range", [1, 5])
      {:ok, [15]}

  **Notes:**
  1. Make sure to use the same store for the linked modules and the main module.
  2. This library does not currently support dependencies between linked modules.
     For example: if Module A links to Module B, and Module B links to the Main module, this will not work.
     Workaround: Link both Module A and B to the Main module directly.

```